src/gui/basegraphgl.h

```diff
@@ -56,12 +56,12 @@ class GraphGLInterface
 protected:
     virtual ~GraphGLInterface() = default;
     virtual void paintFrame(QPainter& painter) const = 0;
-    virtual Node selectNode(const QPointF& pos, bool center) = 0;
+    virtual Node selectNode(const QPointF& pos, bool center, bool addNode = false) = 0;
```

Comment:
do we really need the `bool addNode` flag here?

Reply:
We want this to be overwritten by GraphView's declaration (where we need the flag). Can this be achieved otherwise?

Reply:
yes, but my point here is that we probably don't need the `bool addNode` flag at all.
Intuition says that If I select a node, then it's selected. If it's selected, then it's available on the vector of selected nodes.
In this way, I think the flag just adds more complexity/ambiguity to the code.

Reply:
At first it was needed because otherwise we couldn't display the nodes properly (for the same reason I mentioned bellow). However, since the issue is now resolved, we can now get rid of the flag. 

I also needed to change ``selectedNodes`` from a vector to a map otherwise we'd risk adding the same node multiple times. Since maps don't need to be initialized ``m_maxSelectedNodes`` isn't needed anymore, but it could be a good idea to set an arbitrary limit (of course we could delete it now and re-address this when needed)